iceoryx_utils/include/iceoryx_utils/concurrent/lockfree_queue.hpp

```diff
@@ -48,54 +48,56 @@ class LockFreeQueue
     LockFreeQueue& operator=(LockFreeQueue&&) = delete;
 
     /// @brief returns the capacity of the queue
-    /// threadsafe, lockfree
+    /// @note threadsafe, lockfree
     constexpr uint64_t capacity() const noexcept;
 
     /// @brief tries to insert value in FIFO order, moves the value internally
     /// @param value to be inserted
     /// @return true if insertion was successful (i.e. queue was not full during push), false otherwise
-    /// threadsafe, lockfree
+    /// @note threadsafe, lockfree
     bool tryPush(ElementType&& value) noexcept;
 
     /// @brief tries to insert value in FIFO order, copies the value internally
     /// @param value to be inserted
     /// @return true if insertion was successful (i.e. queue was not full during push), false otherwise
-    /// threadsafe, lockfree
+    /// @note threadsafe, lockfree
     bool tryPush(const ElementType& value) noexcept;
 
     /// @brief inserts value in FIFO order, always succeeds by removing the oldest value
     /// when the queue is detected to be full (overflow)
     /// @param value to be inserted is copied into the queue
     /// @return removed value if an overflow occured, empty optional otherwise
-    /// threadsafe, lockfree
+    /// @note threadsafe, lockfree
     iox::cxx::optional<ElementType> push(const ElementType& value) noexcept;
 
     /// @brief inserts value in FIFO order, always succeeds by removing the oldest value
     /// when the queue is detected to be full (overflow)
     /// @param value to be inserted is moved into the queue if possible
     /// @return removed value if an overflow occured, empty optional otherwise
-    /// threadsafe, lockfree
+    /// @note threadsafe, lockfree
     iox::cxx::optional<ElementType> push(ElementType&& value) noexcept;
 
     /// @brief tries to remove value in FIFO order
     /// @return value if removal was successful, empty optional otherwise
-    /// threadsafe, lockfree
+    /// @note threadsafe, lockfree
     iox::cxx::optional<ElementType> pop() noexcept;
 
     /// @brief check whether the queue is empty
     /// @return true iff the queue is empty
-    /// note that if the queue is used concurrently it might
+    /// @note that if the queue is used concurrently it might
     /// not be empty anymore after the call
     ///  (but it was at some point during the call)
+    /// @note threadsafe, lockfree
     bool empty() const noexcept;
 
     /// @brief get the number of stored elements in the queue
     /// @return number of stored elements in the queue
-    /// note that this will not be perfectly in sync with the actual number of contained elements
+    /// @note that this will not be perfectly in sync with the actual number of contained elements
     /// during concurrent operation but will always be at most capacity
+    /// @note threadsafe, lockfree
     uint64_t size() const noexcept;
 
-  private:
+  protected:

```

Comment:
I am not sure whether AUTOSAR rules out protected variables. If so, it limits useful object oriented design quite a bit I think. I would need to further look into it and we may consider it in the future. If protected methods are not allowed, I consider this rule unusable. Protected variables can be worked around with access methods returning references, but this is just cheating. I would ignore this rule for now because I do not think it should be universal. 

Besides, the variables are only accessible by classes inheriting from it, and those are independent implementations in my opinion which need to consider their own safety rules.

Reply:
I suppose you're talking about `Rule M11-0-1`? Probably slightly outdated, but for reference https://www.autosar.org/fileadmin/user_upload/standards/adaptive/17-03/AUTOSAR_RS_CPP14Guidelines.pdf

Reply:
> Protected variables can be worked around with access methods returning references

I think this then, would be against to Autosar Rule A9-3-1.

I think the main idea of the rule is that, the base class should expose a behavior to its children rather than several raw variables which might get abused/misused easier in the future incarnations of the class. I am not sure how strict this is rule is applied out there in general but I would think it as one of the less nitpicky MISRA/AUTOSAR rules. 

Though I understand your point that this simplifies the code and we are all consenting adults. So if strict compliance with these guidelines is not a goal in the short term, then I am totally OK with ignoring this. I just wanted to throw this on the table.

Reply:
One situation where we quite often use protected is to access some members/method in tests. It might be difficult to change this

Reply:
Ok thanks. Better fix it now, than later ;-)
@marthtz Could you run QAC on this branch tomorrow? I'll try Axivion together with @MatthiasKillat end of this week (off-topic: Will open an issue about our SCA tools of choice early next year, we need to find some solution that fits everyone here)

Reply:
Axivion has to wait, we will roll this out after setting up the process with the whole team. Then we will check what kinds of findings it tends to produces and fix them in separate pull requests on the whole codebase (or by module).